x/auth/legacy/v043/store.go

```diff
@@ -0,0 +1,178 @@
+package v043
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting/exported"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/gogo/protobuf/grpc"
+	"github.com/gogo/protobuf/proto"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+const (
+	delegatorDelegationPath = "/cosmos.staking.v1beta1.Query/DelegatorDelegations"
+	balancesPath            = "/cosmos.bank.v1beta1.Query/AllBalances"
+)
+
+func migrateVestingAccounts(ctx sdk.Context, accounts []types.AccountI, queryServer grpc.Server) ([]types.AccountI, error) {
+	for i := 0; i < len(accounts); i++ {
+		asVesting := vesting(accounts[i])
+		if asVesting == nil {
+			continue
+		}
+
+		account := accounts[i]
+
+		addr := account.GetAddress().String()
+		balance, err := getBalance(
+			ctx,
+			addr,
+			queryServer,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		delegations, err := getDelegatorDelegationsSum(
+			ctx,
+			addr,
+			queryServer,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		asVesting, ok := resetVestingDelegatedBalances(asVesting)
+		if !ok {
+			continue
+		}
+
+		// balance before any delegation includes balance of delegation
+		for _, coin := range delegations {
+			balance = balance.Add(coin)
+		}
+
+		asVesting.TrackDelegation(ctx.BlockTime(), balance, delegations)
+
+		accounts[i] = asVesting.(types.AccountI)
+	}
+
+	return accounts, nil
+}
+
+func vesting(account types.AccountI) exported.VestingAccount {
+	v, ok := account.(exported.VestingAccount)
+	if !ok {
+		return nil
+	}
+
+	return v
+}
+
+func resetVestingDelegatedBalances(evacct exported.VestingAccount) (exported.VestingAccount, bool) {
+	// reset `DelegatedVesting` and `DelegatedFree` to zero
+	df := sdk.NewCoins()
+	dv := sdk.NewCoins()
+
+	switch vacct := evacct.(type) {
+	case *vestingtypes.ContinuousVestingAccount:
+		vacct.DelegatedVesting = dv
+		vacct.DelegatedFree = df
+		return vacct, true
+	case *vestingtypes.DelayedVestingAccount:
+		vacct.DelegatedVesting = dv
+		vacct.DelegatedFree = df
+		return vacct, true
+	case *vestingtypes.PeriodicVestingAccount:
+		vacct.DelegatedVesting = dv
+		vacct.DelegatedFree = df
+		return vacct, true
```

Comment:
```suggestion
	switch vacct := evacct.(type) {
	case *vestingtypes.ContinuousVestingAccount, *vestingtypes.DelayedVestingAccount, *vestingtypes.PeriodicVestingAccount:
		vacct.DelegatedVesting = sdk.NewCoins()
		vacct.DelegatedFree = sdk.NewCoins()
		return vacct, true
```

Reply:
not 100% sure, but I think you can merge the conditions here.

Reply:
Go spec says it can't be done sadly, I would've happily merged your approach otherwise.